commit a38e46cf4907ef47b86bd039c430b2abd1af5712 (HEAD)
Author: Diomidis Spinellis <dds@aueb.gr>
Date:   Fri Nov 1 23:17:38 2019 +0200

    Fix key code reading
    
    - Rewrite statement with undefined behavior
    - Show unknown keys

commit f6ca23e74bf493e379ca63b0be1a20ee3a21572d
Author: Diomidis Spinellis <dds@aueb.gr>
Date:   Fri Nov 1 22:38:07 2019 +0200

    Add header required for strcmp

commit 53b4b1f8570510bd158676957469b2fa05088218
Author: Diomidis Spinellis <dds@aueb.gr>
Date:   Fri Nov 1 22:37:12 2019 +0200

    Fix clash with built-in log function

commit 79bd13b1a56746d3a848444b5d95bb1739e3c3e2
Author: Diomidis Spinellis <dds@aueb.gr>
Date:   Fri Nov 1 22:32:17 2019 +0200

    Include required header file for fd_set


diff --git a/MapReader.cpp b/MapReader.cpp
index aa86f32..02e9715 100644
--- a/MapReader.cpp
+++ b/MapReader.cpp
@@ -105,8 +105,10 @@ void MapReader::LoadMap(const char *filename,EventMapping *map)
 			
 			i = 0;
 
-			while(i<KEY_NAME_MAXLENGTH && ptr[i]!='\0' && ptr[i]!='\n' && (!isspace(ptr[i])))
-				key_name[i] = ptr[i++];
+			while(i<KEY_NAME_MAXLENGTH && ptr[i]!='\0' && ptr[i]!='\n' && (!isspace(ptr[i]))) {
+				key_name[i] = ptr[i];
+				i++;
+			}
 			
 			if(i==KEY_NAME_MAXLENGTH)
 			{
@@ -150,7 +152,7 @@ void MapReader::LoadMap(const char *filename,EventMapping *map)
 				key = Keys::Lookup(key_name);
 				if(key<0)
 				{
-					sprintf(error,"Unknown key name at line %d",line);
+					sprintf(error,"Unknown key name [%s] at line %d", key_name, line);
 					throw Exception("MapReader",error);
 				}
 			}
diff --git a/hid.c b/hid.c
index 42829ee..563b548 100644
--- a/hid.c
+++ b/hid.c
@@ -17,12 +17,14 @@
  * Author: Thibault Kummer <bob@coldsource.net>
  */
 
+#include <sys/select.h>
 #include <hid.h>
 #include <log.h>
 #include <fcntl.h>
 #include <unistd.h>
 #include <dirent.h>
 #include <stdio.h>
+#include <string.h>
 
 #define SYSFS_HIDRAW_CLASS_PATH "/sys/class/hidraw"
 
diff --git a/include/log.h b/include/log.h
index 39c58e4..3358e74 100644
--- a/include/log.h
+++ b/include/log.h
@@ -33,8 +33,8 @@ typedef enum LogLevel {
 
 extern int currentLogLevel;
 
-void log(LogLevel severity, const char* fmt, va_list args);
-//void log(LogLevel severity, const char* fmt, ...);
+void logmsg(LogLevel severity, const char* fmt, va_list args);
+//void logmsg(LogLevel severity, const char* fmt, ...);
 
 void info(const char* fmt, ...);
 void warn(const char* fmt, ...);
diff --git a/log.cpp b/log.cpp
index 8325447..a834abc 100644
--- a/log.cpp
+++ b/log.cpp
@@ -28,18 +28,18 @@ const char* headers[] = {
 	"ERRO",
 };
 
-void log(LogLevel severity, const char* fmt, va_list args) {
+void logmsg(LogLevel severity, const char* fmt, va_list args) {
 	if (severity < currentLogLevel) {
 		vfprintf(stderr, fmt, args);
 		putc('\n', stderr);
 	}
 }
 
-void log(LogLevel severity, const char* fmt, ...)
+void logmsg(LogLevel severity, const char* fmt, ...)
 {
 	va_list args;
 	va_start(args, fmt);
-	log(severity, fmt, args);
+	logmsg(severity, fmt, args);
 	va_end(args);
 
 }
@@ -48,7 +48,7 @@ void info(const char* fmt, ...)
 {
 	va_list args;
 	va_start(args, fmt);
-	log(LOG_INFO, fmt, args);
+	logmsg(LOG_INFO, fmt, args);
 	va_end(args);
 }
 
@@ -56,7 +56,7 @@ void warn(const char* fmt, ...)
 {
 	va_list args;
 	va_start(args, fmt);
-	log(LOG_WARN, fmt, args);
+	logmsg(LOG_WARN, fmt, args);
 	va_end(args);
 }
 
@@ -64,7 +64,7 @@ void error(const char* fmt, ...)
 {
 	va_list args;
 	va_start(args, fmt);
-	log(LOG_ERROR, fmt, args);
+	logmsg(LOG_ERROR, fmt, args);
 	va_end(args);
 }
 
